app/controllers/api/v1/participants_controller.rb

```diff
@@ -2,7 +2,42 @@ module Api::V1
   class ParticipantsController < ApplicationController
 
     def index
-      render json: Participant.all
+      render json: Participant.accepted
+    end
+
+    def create
+      participant = Participant.new(participant_params)
+
+      if participant.save
+        render json: participant
+      else
+        render json: { error: participant.errors.full_messages }
+      end
+    end
+
+    def update
+      unless Participant.statuses.keys.include?(params[:status])
+        render json: { error: 'Invalid status' }
+        return
+      end
+
+      if participant = Participant.find_by_id(params[:id])
```

Comment:
@topdev531  here we could extract the handling of the ActiveRecord::RecordNotFound exception to the controller class

```
  rescue_from ActiveRecord::RecordNotFound, with: :not_found

  def not_found
    render json: { message: "Not Found" }, status: 404
  end
```

And in the action we will only need 
```
        participant = Participant.find(params[:id])
        if participant.pending?
          participant.status = params[:status]
          participant.save
          render json: participant
        else
          render json: { error: "Participant has already been #{participant.status}" }
        end

```